app/javascript/src/js/playlist.js

```diff
@@ -43,23 +29,10 @@ $( document ).ready(function() {
 	};
 
 	SciHistAudioPlaylist.prototype.loadTrack = function(track) {
-		// css (for styling)
-		this.findByRole('track-listing').removeClass("currently-selected");
-		$(track).addClass("currently-selected");
-
-		// data attribute (for identifying the item).
-		this.findByRole('track-listing').attr('data-currently-selected', 'false');
-		$(track).attr('data-currently-selected', 'true');
-
-		this.findByRole('current-track-label').html( $(track).data('title'));
-
-		$(this.audioElement).find("source[type='audio/mpeg']").attr("src", $(track).data('mp3Url'));
-		$(this.audioElement).find("source[type='audio/webm']").attr("src", $(track).data('webmUrl'));
-
-		// Tell HTML audio to load new stuff
-		// See: https://stackoverflow.com/questions/9421505/switch-audio-source-with-jquery-and-html5-audio-tag
-		this.audioElement.pause();
-		this.audioElement.load();
+		var startTime = $(track).data('startTime');
+		if (startTime != -1) {
+			this.audioElement.currentTime = startTime;
+		}
```

Comment:
I don't think you need any JS at all, located in master in simple_ohms_player.js is already code that will catch any link with a `data-ohms-timestamp-s="#{time_in_seconds}"` attribute, and advance the player to that timecode. So you can just make use of that, with such an attribute, and no custom JS at all?

Maybe if the start time is missing (which would be an error condition), simply don't add that attribute to the link at all. Now no need for special -1 handling either?

Reply:
Got rid of the entire file!